lib/commands/network.js

```diff
@@ -76,13 +76,9 @@ commands.toggleFlightMode = async function () {
   });
 };
 
-commands.setGeoLocation = async function (location) {
-  /*
-   * TODO: Implement isRealDevice(). This method fails on
-   * real devices, it should throw a NotYetImplementedError
-   */
-  let cmd = `geo fix ${location.longitude} ${location.latitude}`;
-  return await this.adb.sendTelnetCommand(cmd);
+commands.setGeoLocation = async function(location) {    
+    let cmd = ['am', 'startservice', '-e longitude', location.longitude, '-e latitude', location.latitude, 'io.appium.settings/.LocationService'];
```

Comment:
I spoke too soon. This needs to separate out all the elements. So `-e` and `longitude` need to be separated, for instance.